src/cli/cli_srp_client.cpp

```diff
@@ -349,89 +361,94 @@ otError SrpClient::ProcessService(uint8_t aArgsLength, char *aArgs[])
 
     if (strcmp(aArgs[0], "add") == 0)
     {
-        // `add` <instance-name> <service-name> <port> [priority] [weight] [txt]
+        error = ProcessServiceAdd(aArgsLength, aArgs);
+    }
+    else if (strcmp(aArgs[0], "remove") == 0)
+    {
+        // `remove` <instance-name> <service-name>
 
-        Service *entry = nullptr;
+        const otSrpClientService *service;
 
-        VerifyOrExit(4 <= aArgsLength && aArgsLength <= 7, error = OT_ERROR_INVALID_ARGS);
+        VerifyOrExit(aArgsLength == 3, error = OT_ERROR_INVALID_ARGS);
 
-        for (Service &poolEntry : mServicePool)
+        for (service = otSrpClientGetServices(mInterpreter.mInstance); service != nullptr; service = service->mNext)
         {
-            if (!poolEntry.IsInUse())
+            if ((strcmp(aArgs[1], service->mInstanceName) == 0) && (strcmp(aArgs[2], service->mName) == 0))
             {
-                entry = &poolEntry;
                 break;
             }
         }
 
-        VerifyOrExit(entry != nullptr, error = OT_ERROR_NO_BUFS);
+        VerifyOrExit(service != nullptr, error = OT_ERROR_NOT_FOUND);
 
-        memset(&entry->mService, 0, sizeof(entry->mService));
+        error = otSrpClientRemoveService(mInterpreter.mInstance, const_cast<otSrpClientService *>(service));
+    }
+    else
+    {
+        error = OT_ERROR_INVALID_COMMAND;
+    }
 
-        SuccessOrExit(error = CopyString(entry->mInstanceName, aArgs[1]));
-        entry->mService.mInstanceName = entry->mInstanceName;
+exit:
+    return error;
+}
 
-        SuccessOrExit(error = CopyString(entry->mServiceName, aArgs[2]));
-        entry->mService.mName = entry->mServiceName;
+otError SrpClient::ProcessServiceAdd(uint8_t aArgsLength, char *aArgs[])
+{
+    // `add` <instance-name> <service-name> <port> [priority] [weight] [txt]
 
-        SuccessOrExit(error = ParseAsUint16(aArgs[3], entry->mService.mPort));
+    otSrpClientPoolServiceEntry *entry = nullptr;
+    uint16_t                     size;
+    char *                       string;
+    otError                      error;
 
-        if (aArgsLength >= 5)
-        {
-            SuccessOrExit(error = ParseAsUint16(aArgs[4], entry->mService.mPriority));
-        }
+    VerifyOrExit(4 <= aArgsLength && aArgsLength <= 7, error = OT_ERROR_INVALID_ARGS);
 
-        if (aArgsLength >= 6)
-        {
-            SuccessOrExit(error = ParseAsUint16(aArgs[5], entry->mService.mWeight));
-        }
+    entry = otSrpClientPoolAllocateService(mInterpreter.mInstance);
```

Comment:
Do we allow duplicate entries? Checking would add additional code, so I'm not sure duplicate checking is worth the tradeoff.

Reply:
I agree that checking for duplicate entries may not be worth the code overhead. 

I think the SRP server would reject duplicate service entries (if we try to register same service again) so I expect that  we get an error from server (in the callback) if duplicate entries are added.